mm2src/coins/utxo/utxo_withdraw.rs

```diff
@@ -102,16 +103,20 @@ where
     fn request(&self) -> &WithdrawRequest;
 
     fn signature_version(&self) -> SignatureVersion {
-        match self.sender_address().addr_format {
+        match self.sender_address().addr_format() {
             UtxoAddressFormat::Segwit => SignatureVersion::WitnessV0,
             _ => self.coin().as_ref().conf.signature_version,
         }
     }
 
-    fn prev_script(&self) -> Script {
-        match self.sender_address().addr_format {
-            UtxoAddressFormat::Segwit => Builder::build_p2witness(&self.sender_address().hash),
-            _ => Builder::build_p2pkh(&self.sender_address().hash),
+    #[allow(clippy::result_large_err)]
+    fn prev_script(&self) -> Result<Script, MmError<WithdrawError>> {
+        match self.sender_address().addr_format() {
+            UtxoAddressFormat::Segwit => match Builder::build_p2wpkh(self.sender_address().hash()) {
+                Ok(script) => Ok(script),
+                Err(e) => MmError::err(WithdrawError::InternalError(e.to_string())),
+            },
+            _ => Ok(Builder::build_p2pkh(self.sender_address().hash())),
```

Comment:
Please cover all variants explicitly, for instance if we were to add taproot address in the future the compiler won't ask us to fix this and will build it as `p2pkh` if the dev forgets to check it.